src/pytypes/onefuzztypes/models.py

```diff
@@ -476,15 +494,18 @@ class WorkerDoneEvent(BaseModel):
     stdout: str
 
 
-class WorkerEvent(BaseModel):
-    event: Union[WorkerDoneEvent, WorkerRunningEvent]
+class WorkerEvent(EnumModel):
+    done: Optional[WorkerDoneEvent]
+    running: Optional[WorkerRunningEvent]
 
 
 class NodeStateUpdate(BaseModel):
     state: NodeState
 
 
-NodeEvent = Union[WorkerEvent, NodeStateUpdate]
+class NodeEvent(EnumModel):
+    worker_event: Optional[WorkerEvent]
```

Comment:
Now, types like `NodeEvent` have a field for each variant of the (logical) discriminated union, instead of being represented by a `Union` (which is not a pydantic model, or even an instantiable class). A root validator in the superclass ensures that exactly one of the optional fields is non-`None`.

The motivation for moving away from this we can't do checks like `isinstance(foo, Union[A, B])`. We would need do this when we are scrutinizing a type that we would know to have type `Union[Union[A, B], C]`. So, any type that is logically just a `Union` previously needed to be wrapped in a model with a single artificial field, whose value is a union. This was very strange on the Rust side.

Reply:
Eventually we expect pydantic to grow internally-tagged discriminated unions, and there will be much less mismatch between the types.